canonical/ubuntu-pro-client
# Binary package rename (SC-1623)

## Why is this needed?

This PR renames the binary packages created from ubuntu-advantage-tools as agreed. The new package names are "ubuntu-pro-client" and "ubuntu-pro-image-auto-attach".

This PR is packaging heavy and deserves extra scrutiny there.

This PR also adjusts our tests to use the new packages.

A subsequent PR will also include a manual test script for testing the transition

## Test Steps
Test adding this ppa and upgrading: `ppa:orndorffgrant/proclient-rename-test`

A test that can only be done with debs and not with a proper apt source of the full ubuntu-advantage-tools set of binaries:
1. launch a lxd container
2. copy over just ubuntu-pro-client.deb
3. `apt install ubuntu-pro-client.deb`
That should fail saying that ubuntu-advantage-tools >=31 is also required

## Checklist


## Does this PR require extra reviews?
